subprojects/docs/docs.gradle

```diff
@@ -447,17 +280,109 @@ tasks.register("checkstyleApi", Checkstyle) {
     reports.xml.destination = file("$checkstyle.reportsDir/checkstyle-api.xml")
 }
 
-def distDocs = tasks.register("distDocs", Docbook2Xhtml) {
-    dependsOn userguideAsciidoc
-    source new File(asciidocOutputDir, 'getting_started.xml')
-    destFile = new File(distDocsDir, 'getting-started.html')
-    stylesheetName = 'standaloneHtml.xsl'
+def distDocs = tasks.register("distDocs", AsciidoctorTask) {
+    sourceDir = userguideSrcDir
+    sources { include 'getting-started.adoc' }
+    outputDir = distDocsDir
+    separateOutputDirs = false
+    options doctype: 'book'
+    backends 'html5'
+}
+
+asciidoctorj {
+    noDefaultRepositories = true
+    version = '1.5.7'
 }
 
-def userguide = tasks.register("userguide") {
-    dependsOn userguideHtml, userguideSingleHtml, userguidePdf
-    description = 'Generates the userguide'
+def userguideAsciidoc = tasks.register("userguideAsciidoc", AsciidoctorTask) {
```

Comment:
Also consider:

 - [ ] Rename 'userguide' tasks to 'userManual...'

Reply:
Or perhaps 'usermanual...' so that the shorthand notation is the same as it was before. I favour 'userManual...' myself.

Reply:
I also favor userManual everywhere, but I want to avoid renaming things just for now. Things like CI jobs may use `userguide` and I don't want to have to keep around both userguide and userManual until we do some refactoring.

Reply:
Fair enough.